Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -1005,13 +1043,137 @@ namespace Terrain
         m_raytracingNormalsBuffer->Unmap();
     }
 
+    void TerrainMeshManager::UpdateCandidateSectors()
+    {
+        // Gather a list of all sectors that could render based on their status, lod, and camera position.
+
+        float maxDistanceSq = m_config.m_firstLodDistance * m_config.m_firstLodDistance;
+        uint32_t nextLodSectorCount = m_1dSectorCount * 2; // The number of this lod's sectors that would fit into the next lod's space.
+        AZStd::vector<bool> previousSelectedSectors;
+
+        m_candidateSectors.clear();
+
+        for (uint32_t lodLevel = 0; lodLevel < m_sectorLods.size(); ++lodLevel)
+        {
+            auto& lodGrid = m_sectorLods.at(lodLevel);
+
+            // Each sector in an LOD is half the size of a sector in the next LOD in each direction, so 4 sectors
+            // in this LOD equal one sector in the next LOD. Construct a grid of bools for each sector in this
+            // LOD that covers the entire space of the next LOD, and mark everything to false. As sectors in
+            // this LOD are drawn, mark appropriate locations in the grid as true. When processing the next
+            // LOD, each of the next LOD's sectors will look up the 4 entries in this that represent quadrants
+            // of that sector to determine whether to draw nothing, specific quadrants, or the entire sector.
+
+            AZStd::vector<bool> selectedSectors(nextLodSectorCount * nextLodSectorCount, false);
+
+            Vector2i selectedSectorStartCoord{ 0, 0 };
+            if (lodLevel == m_sectorLods.size() - 1)
+            {
+                // There is no next lod, so just use this one's start coord to avoid lots of checks in the for loop.
+                selectedSectorStartCoord = m_sectorLods.at(lodLevel).m_startCoord;
+            }
+            else
+            {
+                // This is the start coord of the next LOD in the current LOD's scale.
+                selectedSectorStartCoord = m_sectorLods.at(lodLevel + 1).m_startCoord * 2;
+            }
+
+            for (uint32_t sectorIndex = 0; sectorIndex < lodGrid.m_sectors.size(); ++sectorIndex)
+            {
+                Sector& sector = lodGrid.m_sectors.at(sectorIndex);
```

Comment:
Nitpick: Could this be a `const Sector&` ?

Reply:
Nice catch -  I'll do that if AR fails again, otherwise I'll get it in the next PR